app/Models/BeatmapDiscussionPost.php

```diff
@@ -163,6 +163,31 @@ public function user()
         return $this->belongsTo(User::class, 'user_id');
     }
 
+    /**
+     * Whether a post can be edited/deleted.
+     *
+     * When a discussion is resolved, the posts preceeding the resolution are locked.
+     * Posts after the resolution are not locked, unless the issue is re-opened and resolved again.
+     *
+     * @return bool
+     */
+    public function canEdit()
+    {
+        if ($this->system) {
+            return false;
+        }
+
+        // The only system post type currently implemented is 'resolved', so we're making the assumption
+        // the next system post is always going to be either a resolve or unresolve.
+        // This will have to be changed if more types are added.
+        $systemPost = static::where('system', true)
+            ->where('id', '>', $this->id)
+            ->where('beatmap_discussion_id', $this->beatmap_discussion_id)
+            ->first();
```

Comment:
Shouldn't it find the last resolving post instead of earliest state change post?

1
2
3-resolve
4
5-unresolve
6
7-resolve
8
9-unresolve
10
11

In this case only 8+ should be editable, shouldn't it? With current check 4 is editable.